File: 2_1D_Modell/2_7_SW_Old___TBD/Matlab_WS/Library/CFilterData.m
classdef CFilterData < handle
    properties
        mTime
        mPhi
        mPhi__d
        mPhi__dd
    end
    
    methods
        function ctor = CFilterData()
            ctor.mTime    = [];
            ctor.mPhi     = [];
            ctor.mPhi__d  = [];
            ctor.mPhi__dd = [];
        end
        function addData(this, time, phi, phi__d, phi__dd)
            this.mTime      = [this.mTime, double(time)];
            this.mPhi       = [this.mPhi, double(phi)];
            this.mPhi__d    = [this.mPhi__d, double(phi__d)];
            this.mPhi__dd   = [this.mPhi__dd, double(phi__dd)];
        end
    end
    
end

File: 2_1D_Modell/2_7_SW_Old___TBD/Eclipse_WS/CuBa/Basic/source/CFilterData.cpp
#include "CFilterData.h"

CFilterData::CFilterData(Float32 time,
						 Float32 phi,
						 Float32 phi__d,
						 Float32 phi__dd) : mTime(time),
						 	 	 	 	    mPhi(phi),
											mPhi__d(phi__d),
											mPhi__dd(phi__dd)
{

}
Float32 CFilterData::getTime() const
{
	return mTime;
}
Float32 CFilterData::getPhi() const
{
	return mPhi;
}
Float32 CFilterData::getPhi__d() const
{
	return mPhi__d;
}
Float32 CFilterData::getPhi__dd() const
{
	return mPhi__dd;
}
